include/mfmg/common/lanczos.templates.hpp

```diff
@@ -29,6 +31,47 @@
 namespace mfmg
 {
 
+namespace internal
+{
+template <typename VectorType>
+void details_set_initial_guess(VectorType &initial_guess, int seed)
+{
+  // Modify initial guess with a random noise by multiplying each entry of the
+  // vector with a random value from a uniform distribution. This specific
+  // procedure guarantees that zero entries of the vector stay zero, which is
+  // important for situations where they are associated with constrained dofs in
+  // Deal.II
```

Comment:
That's because constrained dofs introduce spurious eigenvalues, so we just use an initial guess that has zeros on the constrained dofs. When we deflate, we want to keep the entries to zero.